src/main/java/bio/terra/flight/dataset/ingest/IngestSetupStep.java

```diff
@@ -49,6 +49,38 @@
     private DatasetService datasetService;
     private BigQueryPdao bigQueryPdao;
 
+    static void validateSourceUri(String sourcePath) {
+        IngestUtils.GsUrlParts gsParts = IngestUtils.parseBlobUri(sourcePath);
+
+        // Bucket wildcards are never supported.
+        if (gsParts.getBucket().indexOf('*') > -1) {
+            throw new InvalidUriException("Bucket wildcards are not supported: " + sourcePath);
+        }
+
+        int globIndex = gsParts.getPath().indexOf('*');
+        if (globIndex == -1) {
+            // If the user is trying to ingest a single file, verify it exists.
+            // TODO: If we're OK letting BQ hit and return the "not found" error in the wildcard case,
```

Comment:
I think we should handle the bigquery error case and not try to prevent it from happening here. There is a window between this step and when BQ actually runs its operation. The state tested here could change in that window, so we have to handle it there anyway.